src/wallet/rpcdump.cpp

```diff
@@ -1261,6 +1262,8 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
             if (!pwallet->GetPubKey(id, temp) && !pwallet->AddWatchOnly(GetScriptForRawPubKey(pubkey), timestamp)) {
                 throw JSONRPCError(RPC_WALLET_ERROR, "Error adding address to wallet");
             }
+            pwallet->AddKeyOrigin(pubkey, import_data.key_origins[id]);
```

Comment:
Should this be called when `id` is not a member of `import_data.key_origins`? The non-descriptor imports won't have this map entries created.

Reply:
It should check that. I've added the check and a test for this.